p2p/conn/connection.go

```diff
@@ -452,8 +484,10 @@ FOR_LOOP:
 			c.Logger.Debug("Receive Ping")
 			c.pong <- struct{}{}
 		case packetTypePong:
-			// do nothing
 			c.Logger.Debug("Receive Pong")
+			if c.pongTimer != nil {
```

Comment:
won't this race with sendRoutine? If the ping timer fires around when we receive a pong

Reply:
it probably will. what's the solution? mutex and another channel?

Reply:
also interesting why go racer does not say anything...